cmd/argocd/commands/project.go

```diff
@@ -999,3 +1001,95 @@ func NewProjectEditCommand(clientOpts *argocdclient.ClientOptions) *cobra.Comman
+			argocd proj add-destination-service-account PROJECT SERVER NAMESPACE SERVICE_ACCOUNT
+		`),
+		Run: func(c *cobra.Command, args []string) {
+			ctx := c.Context()
+
+			if len(args) != 3 {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			projName := args[0]
+			namespace := args[2]
+			serviceAccount := args[3]
+			destinationServiceAccount := buildApplicationDestinationServiceAccount(args[1], namespace, serviceAccount)
+			conn, projIf := headless.NewClientOrDie(clientOpts, c).NewProjectClientOrDie()
+			defer argoio.Close(conn)
+
+			proj, err := projIf.Get(ctx, &projectpkg.ProjectQuery{Name: projName})
+			errors.CheckError(err)
+
+			for _, dest := range proj.Spec.DestinationServiceAccounts {
+				dstServerExist := destinationServiceAccount.Server != "" && dest.Server == destinationServiceAccount.Server
+				dstServiceAccountExist := destinationServiceAccount.DefaultServiceAccount != "" && dest.DefaultServiceAccount == destinationServiceAccount.DefaultServiceAccount
+				if dest.Namespace == destinationServiceAccount.Namespace && dstServerExist && dstServiceAccountExist {
+					log.Fatal("Specified destination service account is already defined in project")
+				}
+			}
+			proj.Spec.DestinationServiceAccounts = append(proj.Spec.DestinationServiceAccounts, destinationServiceAccount)
+			_, err = projIf.Update(ctx, &projectpkg.ProjectUpdateRequest{Project: proj})
+			errors.CheckError(err)
+		},
+	}
+	return command
+}
+
+// NewProjectRemoveDestinationCommand returns a new instance of an `argocd proj remove-destination-service-account` command
+func NewProjectRemoveDestinationServiceAccountCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
+	var command = &cobra.Command{
+		Use:   "remove-destination-service-account PROJECT SERVER NAMESPACE SERVICE_ACCOUNT",
+		Short: "Remove default destination service account from the project",
+		Example: templates.Examples(`
+			# Remove the destination service account (SERVICE_ACCOUNT) from the specified destination (SERVER and NAMESPACE combination) on the project with name PROJECT
+			argocd proj remove-destination-service-account PROJECT SERVER NAMESPACE SERVICE_ACCOUNT
+		`),
+		Run: func(c *cobra.Command, args []string) {
+			ctx := c.Context()
+
+			if len(args) != 3 {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			projName := args[0]
+			server := args[1]
+			namespace := args[2]
+			serviceAccount := args[3]
+			conn, projIf := headless.NewClientOrDie(clientOpts, c).NewProjectClientOrDie()
+			defer argoio.Close(conn)
+
+			proj, err := projIf.Get(ctx, &projectpkg.ProjectQuery{Name: projName})
+			errors.CheckError(err)
+
+			index := -1
+			for i, dest := range proj.Spec.DestinationServiceAccounts {
+				if dest.Namespace == namespace && dest.Server == server && dest.DefaultServiceAccount == serviceAccount {
+					index = i
+					break
+				}
+			}
+			if index == -1 {
```

Comment:
Can this code be simplified.
```
if dest.Namespace == namespace && dest.Server == server && dest.DefaultServiceAccount == serviceAccount {
	proj.Spec.Destinations = append(proj.Spec.Destinations[:index], proj.Spec.Destinations[index+1:]...)
        _, err = projIf.Update(ctx, &projectpkg.ProjectUpdateRequest{Project: proj})
        errors.CheckError(err)
        break;
} else {
  log.Fatal("Specified destination service account does not exist in project")
}
```

Reply:
Consider using `slices.DeleteFunc` https://pkg.go.dev/golang.org/x/exp/slices#DeleteFunc to delete the element from the slice.
```
originalLength = len(proj.Spec.DestinationServiceAccounts)
proj.Spec.DestinationServiceAccounts = slices.DeleteFunc(proj.Spec.DestinationServiceAccounts, func( destServiceAccount ApplicationDestinationServiceAccount) { return destServiceAccount.Namespace == namespace && destServiceAccount.Server == server && dest.DefaultServiceAccount == serviceAccount }
if (originalLength != len(proj.Spec.DestinationServiceAccounts)){
    _, err = projIf.Update(ctx, &projectpkg.ProjectUpdateRequest{Project: proj})
   errors.CheckError(err)
} else {
  log.Fatal("Specified destination service account does not exist in project")
}
```